src/common/options.h

```diff
@@ -38,11 +39,11 @@ class Options {
       options_[it.first.as<std::string>()] = YAML::Clone(it.second);
   }
 
-  void merge(Ptr<Options> options) { merge(options->getOptions()); }
+  void merge(Ptr<Options const> options) { merge(options->getOptions()); }
```

Comment:
I am very much in favor of const methods. But const pointers may be tricky. Can a Ptr<X> be assigned to a Ptr<X const>? If so, great.

Reply:
Coming from boost: "In particular, shared_ptr<T> is implicitly convertible to shared_ptr<T const> ...". See https://www.boost.org/doc/libs/1_61_0/libs/smart_ptr/shared_ptr.htm. So yes, according to the documentation. Anything else would be complete bonkers anway, in my opinion.